src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -39,16 +54,20 @@ void uct_rdmacm_cm_ep_error_cb(uct_rdmacm_cm_ep_t *cep,
     }
 
     ucs_assert(status != UCS_OK);
-    cep->flags |= UCT_RDMACM_CM_EP_FAILED;
+    /* Do not set local EP failed on normal disconnect event */
+    if (status != UCS_ERR_CONNECTION_RESET) {
+        cep->flags |= UCT_RDMACM_CM_EP_FAILED;
+    }
+
     cep->status = status;
 
-    if (cep->flags & UCT_RDMACM_CM_EP_GOT_CONNECT) {
+    if (cep->flags & UCT_RDMACM_CM_EP_CONNECTED) {
```

Comment:
pls add code comments for each case
e.g "already connected so call disconnect callback"